### PR TITLE
MM-60387: Bump Go version to 1.22

### PR DESCRIPTION
**Pull request by agnivade**
The first part is to just bump the build image. Once merged to master, it will tag a new image. The next step is to bump up everything else.

https://mattermost.atlassian.net/browse/MM-60387
```release-note
NONE
```

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]

**Comment by agnivade** [comment]
@toninis - This is ready for review.

**Review by toninis — APPROVED** [review]
Let's go GO 🚀

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
     - name: Setup go
       uses: actions/setup-go@v2
       with:
-        go-version: '1.21'
+        go-version: '1.22'
```
[the remark this answers is not in the document]
Comment (agnivade) [reply .github/workflows/codeql-analysis.yml:45]: Ah yes good point.

**Pull request merged**